src/btree/bt_curnext.c

```diff
@@ -499,8 +504,12 @@ __cursor_row_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_t *skip
          * If an upper bound has been set ensure that the key is within the range, otherwise early
          * exit.
          */
-        if (F_ISSET(cursor, WT_CURSTD_BOUND_UPPER))
-            WT_RET(__wt_btcur_bounds_early_exit(session, cbt, true, key_out_of_bounds));
+        ret = __wt_btcur_bounds_early_exit(session, cbt, true, key_out_of_boundsp);
+
+        if (ret == WT_NOTFOUND) {
```

Comment:
This should be:
```
if ((ret =__wt_btcur_bounds_early_exit(session, cbt, true, key_out_of_boundsp)) == WT_NOTFOUND) {

} else 
    WT_RET(ret);
```
Same for other usages.

Reply:
Good point, implementing that now!